openedx-unsupported/edx-analytics-dashboard
# [SE-3950] Add a waffle flag to ignore the accept-language header.

This pull request adds a waffle flag that makes the analytics system ignore the 'Accept-Language' header. The rationale is that while the Accept-Language header is presumed to communicate the user's preferred language, in practice many, perhaps most users do not set this value explicitly in their settings. This is a problem for installations that do not target English users as their primary audience.

When this flag is enabled, LANGUAGE_CODE is used to determine which translation to serve. An existing or future created cookie setting language preference will still work as expected.

**JIRA tickets**: https://openedx.atlassian.net/browse/OSPR-5367

**Dependencies**: None

**Merge deadline**: None

**Testing instructions**:

1. Clone the repository if you have not done so already.
2. Create a virtualenv with Python 3.5 and activate it.
3. Follow the installation instructions in the README.
4. Set ALLOWED_HOSTS to ['*'] to make things easier on yourself while testing.
5. Change the LANGUAGE_CODE setting to a language you'd prefer.
6. Use runserver. Observe that the language header is still respected.
7. Set the waffle flag with `./manage.py waffle_flag analytics_dashboard.ignore_accept_language --everyone --create`
8. Reload the page and verify the language has changed.

**Author notes and concerns**:

This does introduce the edx-toggle library, adding this waffle flag as the first documented to be used in this manner. It does not go back and add documentation for every existing waffle flag, which should probably be done at some point.

**Reviewers**
